packages/cli-kit/src/public/node/ruby.test.ts

```diff
@@ -47,138 +54,176 @@ describe('execCLI', () => {
+      const gemfilePath = joinPath(cli2Directory, 'Gemfile')
+
+      // When
+      await execCLI2(['args'], {
+        token: 'token_0000_1111_2222_3333',
+        directory: './directory',
+      })
+
+      // Then
+      validateBudleExec(execSpy, gemfilePath, joinPath(cli2Directory, 'bin', 'shopify'))
+      await validateGemFileContent(gemfilePath, !isBundled, isWindows)
     })
-
-    // Teardown
-    process.env = originalEnv
   })
 })
+
+function mockBundledCLI2(cli2Directory: string, windows: boolean) {
+  vi.spyOn(local, 'isShopify').mockResolvedValue(false)
+  vi.spyOn(pathConstants.directories.cache.vendor, 'path').mockReturnValue(cli2Directory)
+  mockRubyEnvironment()
+  mockPlatformAndArch(windows)
+  return vi.spyOn(system, 'exec')
+}
+
+async function mockEmbeddedCLI2(cli2Directory: string, windows: boolean, existingWindowsDependency = false) {
+  vi.spyOn(local, 'isShopify').mockResolvedValue(true)
+  vi.spyOn(file, 'findPathUp').mockResolvedValue(cli2Directory)
+  mockRubyEnvironment()
+  mockPlatformAndArch(windows)
+  await createGemFile(cli2Directory, existingWindowsDependency)
+  return vi.spyOn(system, 'exec')
+}
+
+function mockRubyEnvironment() {
+  vi.mocked(captureOutput).mockResolvedValueOnce('2.7.5')
+  vi.mocked(captureOutput).mockResolvedValueOnce('2.4.0')
+}
+
+function mockPlatformAndArch(windows: boolean) {
+  if (windows) {
+    vi.mocked(platformAndArch).mockReturnValue({platform: 'windows', arch: 'x64'})
+  } else {
+    vi.mocked(platformAndArch).mockReturnValue({platform: 'darwin', arch: 'x64'})
+  }
+}
+
+async function createGemFile(cli2Directory: string, existingWindowsDependency: boolean) {
+  const gemfilePath = joinPath(cli2Directory, 'Gemfile')
+  let content = "source 'https://rubygems.org'\n"
+  if (existingWindowsDependency) content = content.concat("gem 'wdm', '>= 0.1.0'")
+  await file.touchFile(gemfilePath)
+  await file.appendFile(gemfilePath, content.concat('\n'))
+}
+
+function validateBudleExec(execSpy: SpyInstance, gemFilePath: string, execPath = 'shopify') {
+  expect(execSpy).toHaveBeenLastCalledWith('bundle', ['exec', execPath, 'args'], {
+    stdio: 'inherit',
+    cwd: './directory',
+    env: {
+      ...process.env,
+      SHOPIFY_CLI_STOREFRONT_RENDERER_AUTH_TOKEN: undefined,
+      SHOPIFY_CLI_ADMIN_AUTH_TOKEN: undefined,
+      SHOPIFY_CLI_STORE: undefined,
+      SHOPIFY_CLI_AUTH_TOKEN: 'token_0000_1111_2222_3333',
+      SHOPIFY_CLI_RUN_AS_SUBPROCESS: 'true',
+      BUNDLE_GEMFILE: gemFilePath,
+    },
+  })
+}
+
+async function validateGemFileContent(gemfilePath: string, bundled: boolean, windows: boolean) {
+  expect(file.fileExists(gemfilePath)).toBeTruthy()
+  const gemContent = await file.readFile(gemfilePath, {encoding: 'utf8'})
+  expect(gemContent).toContain("source 'https://rubygems.org'")
+  if (bundled) expect(gemContent).toContain(`gem 'shopify-cli', '${RubyCLIVersion}'`)
+  const windowsDepency = "gem 'wdm', '>= 0.1.0'"
```

Comment:
Can we have this in a constant so that we can update its value without breaking the tests?